lib/plsql/jdbc_connection.rb

```diff
@@ -5,21 +5,31 @@
   # ojdbc6.jar or ojdbc5.jar file should be in JRUBY_HOME/lib or should be in ENV['PATH'] or load path
 
   java_version = java.lang.System.getProperty("java.version")
-  ojdbc_jar = if java_version =~ /^1.5/
+  ojdbc_jars = if java_version =~ /^1.5/
     "ojdbc5.jar"
-  elsif java_version >= '1.6'
+  elsif java_version =~ /^1.6/
     "ojdbc6.jar"
+  elsif java_version >= "1.7"
+    # Oracle 11g client ojdbc6.jar is also compatible with Java 1.7
+    # Oracle 12c client provides new ojdbc7.jar
+    %w(ojdbc7.jar ojdbc6.jar)
   else
     nil
   end
 
-  unless ENV_JAVA['java.class.path'] =~ Regexp.new(ojdbc_jar)
+  if ojdbc_jars && ENV_JAVA["java.class.path"] !~ Regexp.new(ojdbc_jars.join("|"))
```

Comment:
`join` doesn't work on plain string so I guess this would currently fail for Java < 1.7.